### ISSUE TITLE
Cosmos: Throw a detailed exception for unsupported options when translating Regex.IsMatch

### ISSUE THREAD

**Issue by AndriySvyryd** [dotnet/efcore#28139]
See https://github.com/dotnet/efcore/pull/28121

**Comment by smitpatel** [comment]
Not really blocked, the method translator should throw exception from itself.

**Comment by roji** [comment]
Don't we avoid doing that since it would block client eval?

**Comment by smitpatel** [comment]
Depends if we want to do client eval of certain things.

**Comment by roji** [comment]
O, I'll bring this up in design, possibly with a proposal for #26410.

**Comment by roji** [comment]
Just to figure out how we see things... If we're OK with throwing in method translators, #26410 becomes much less important, maybe even closeable (note it's already in the milestone).

I still think that ideally we just do #26410, e.g. by having translator return some NotTranslatableExpression (which wraps details); and then we don't have to discuss throwing because that always allows client eval and provides nicer exceptions. But we can discuss all that.

**Comment by Marusyk** [comment]
I'll prepare a PR for this

**Comment by roji** [comment]
@Marusyk sure, but we'll need to decide what we want to do with unsupported options (design discussion pending).

**Comment by roji** [comment]
Design decision: we'll do #26410 at some point, but for now it should be fine to just return null for unsupported options, so the user will just get the generic "cannot translate" exception. @Marusyk this should unblock you for working on this issue.

**Comment by Marusyk** [comment]
So, should I return the previous version with [HashSet](https://github.com/dotnet/efcore/pull/28121/files/b8cba0c574c3295aa622a7cc71a954a446f43e83#diff-ddd4f5b756ddfba1b854430190d2fe812fc0bed705ff3263e9d05074a9d24a46R22)?

> "Constantize" regex options via EvaluatableFilter
I'm not sure that understand this task. Do I need to implement CosmosEvaluatableExpressionFilter?

**Comment by roji** [comment]
> So, should I return the previous version with [HashSet](https://github.com/dotnet/efcore/pull/28121/files/b8cba0c574c3295aa622a7cc71a954a446f43e83#diff-ddd4f5b756ddfba1b854430190d2fe812fc0bed705ff3263e9d05074a9d24a46R22)?

Nope, I don't see any reason for doing so. The current code does the right thing (for now) - if there's any unsupported option, it just returns null.

> Do I need to implement CosmosEvaluatableExpressionFilter?

Yeah, @smitpatel can probably offer some more assistance if needed.

**Comment by roji** [comment]
@Marusyk do you intend to work on this?

**Comment by Marusyk** [comment]
Unfortunately, I don't have time to investigate EvaluatableExpressionFilter.

**Comment by roji** [comment]
Discussed with @smitpatel and the constantization of the RegexOptions parameters would require some infra work, so we'll leave that for now.
